### ISSUE TITLE
Enable binary acceptance test driver 

### ISSUE THREAD

**Issue by kmoe** [integrations/terraform-provider-github#373]
Terraform plugin SDK [v1.7.0](https://github.com/hashicorp/terraform-plugin-sdk/blob/master/CHANGELOG.md#170-february-12-2020) includes a new binary acceptance test driver feature, which makes acceptance tests runs more realistic by running `terraform` CLI commands during tests, instead of simulating them using old SDK code. 

Please see https://github.com/hashicorp/terraform-plugin-sdk/pull/262 for more details about this feature, and https://github.com/terraform-providers/terraform-provider-random/pull/99 for an example of how to switch the feature on.

Parking this ticket here for now, as this needs https://github.com/terraform-providers/terraform-provider-github/pull/372. After https://github.com/terraform-providers/terraform-provider-github/pull/372 is merged I'll submit a PR enabling the binary driver for review.

**Comment by benj-fletch** [comment]
Hi @kmoe, thanks a lot for bringing this feature to us, it sounds like a reasonable idea.
#372 has now been merged, so you should be free to go ahead and work on this. :)

I would also like @jcudit to have sight of this since there is a lot of space being done in the testing area at the moment.

**Comment by jcudit** [comment]
Hi @kmoe, please mention me on the PR when it is ready.  Happy to help with reviewing and testing the change 😄

**Comment by jcudit** [comment]
@kmoe can we align this effort with another one that is in the works?  There is a [series of PRs](https://github.com/terraform-providers/terraform-provider-github/issues/83#issuecomment-583940323) that have their tests deferred but are now needed to unblock progress.

Would you be willing to write a few of the missing tests as an example that future contributors can model going forward?
